README.md

```diff
@@ -53,6 +53,12 @@ You will also find detailed [guides](./docs/Guides.md) and [frequently asked que
 
 ### Prerequisite: Bitcoin Core
 
+We chose to rely as much as possible on bitcoin core, to interface with and monitor the blockchain and also to manage onchain funds (eclair does not include an onchain wallet, channel opening transactions are funded by your bitcoin core node, and channel closing transactions return funds to your bitcoin core node).
+
+This means that we have strong requirements on how your bitcoin node is configured (see below), and that you must backup your bitcoin core wallet as well as your eclair node (see https://github.com/ACINQ/eclair/edit/master/README.md#configure-bitcoin-core-wallet)
+
+But is also means that instead of re-implemeting them, we benefit from all the verifications and optimisations (including fee management with RBF/CPFP, ...) that are provided by bitcoin core, which is much better from a security point of view. We did have to implement some of the bitcoin protocol, but in the context of verifying data provided by bitcoin core.
```

Comment:
Out of curiosity, is there any part in Eclair that you'd still consider risky to some degree because it re-implements bitcoin logic? Perhaps any sort of re-parsing of blocks, or did you manage to avoid all of that?

Reply:
Definitely a strong point to rely on battle tested code as much as possible.

Reply:
We still have a utility function that verifies that an unconfirmed transaction is valid (have a look at occurrences of `Transaction.correctlySpends`), where we could potentially deviate from `bitcoind` which would create issues. However, this function is never used to validate transactions that are included in blocks, we only use it when collaboratively building a transaction (e.g. when validating remote signatures of a commit tx or a dual-funded tx). AFAICT, this should never make an `eclair` node halt. The only risks that I see are that we'd either:

1. Accept a transaction that would never confirm because it's actually invalid: that means we would have a channel stuck in the `unconfirmed` state, which doesn't create any risks as long as you don't use 0-conf (if you do, you have a trust assumption on your peer anyway)
2. Reject a transaction that is actually valid: that could lead to eclair rejecting channel funding attempts that should work, which doesn't incur any loss of funds either, but could be pretty annoying 

We also obviously also have custom code to deserialize txs and blocks into scala objects: if we were unable to deserialize some txs or blocks, it could lead to partial denial of service. But we don't run any validation on those objects, which makes the attack surface pretty low, but there is still a potential risk there.

Reply:
Sounds good. Especially the part where you are not forcing every block seen through your own validation code 👍